tests/test_ilastik/test_workflows/test_workflow_listing.py

```diff
@@ -0,0 +1,60 @@
+import ilastik.workflows
+import ilastik.workflow
+
+from unittest.mock import Mock
+import pytest
+
+
+@pytest.mark.parametrize(
+    "workflowclasses,subclasses,expected",
+    [
+        ([Mock(workflowName="Test", __name__="Test", workflowDisplayName=None)], [], (("Test", "Test"),)),
+        ([Mock(workflowName=None, __name__="TestWorkflow", workflowDisplayName=None)], [], (("Test", "Test"),)),
+        (
+            [Mock(workflowName="Test", __name__="Test", workflowDisplayName="TestDisplay")],
+            [],
+            (("Test", "TestDisplay"),),
+        ),
+        (
+            [
+                Mock(workflowName="1", __name__="1", workflowDisplayName=None),
+                Mock(workflowName="2", __name__="2", workflowDisplayName="2Disp"),
+            ],
+            [],
+            (
+                ("1", "1"),
+                ("2", "2Disp"),
+            ),
+        ),
+        ([], [Mock(workflowName="Test", __name__="Test", workflowDisplayName=None)], (("Test", "Test"),)),
+        ([], [Mock(workflowName="TestBase", __name__="Test", workflowDisplayName=None)], tuple()),
+        ([], [Mock(workflowName="Test", __name__="Test", workflowDisplayName=None, auto_register=False)], tuple()),
+        (
+            [
+                Mock(workflowName="1", __name__="1", workflowDisplayName=None),
+            ],
+            [
+                Mock(workflowName="2", __name__="2", workflowDisplayName="2Disp"),
+            ],
+            (
+                ("1", "1"),
+                ("2", "2Disp"),
+            ),
+        ),
+        (
+            [Mock(workflowName="Test", __name__="Test", workflowDisplayName=None)],
+            [Mock(workflowName="Test", __name__="Test", workflowDisplayName=None)],
+            (("Test", "Test"),),
+        ),
+    ],
+)
+def test_listing(monkeypatch, workflowclasses, subclasses, expected):
```

Comment:
Awesome :)
Maybe enclose test cases with `# fmt: off` because it's hard to parse otherwise

Reply:
hmm, what do you mean by hard to parse? Wold you prefer a single line per test case?